File: auth-server/src/test/java/no/vicx/authserver/UserTestUtils.java
package no.vicx.authserver;

import no.vicx.database.user.UserImage;
import no.vicx.database.user.VicxUser;
import org.springframework.http.MediaType;

public final class UserTestUtils {
    private UserTestUtils() {
    }

    public static final String DEFAULT_USERNAME_IN_TEST = "user1";
    public static final String EXISTING_USERNAME = "~username~";
    public static final String NON_EXISTING_USERNAME = "user2";

    public static UserImage createUserImageInTest() {
        return new UserImage(new byte[]{1, 2, 3}, MediaType.IMAGE_PNG_VALUE);
    }

    public static VicxUser createUserInTest(UserImage userImage) {
        return new VicxUser(
                EXISTING_USERNAME, "~password~", "~name~", "~email~", userImage);
    }
}

File: auth-server/src/test/kotlin/no/vicx/authserver/UserTestUtils.kt
package no.vicx.authserver

import no.vicx.authserver.config.DefaultUserProperties
import no.vicx.database.user.UserImage
import no.vicx.database.user.VicxUser
import org.springframework.http.MediaType

object UserTestUtils {
    const val DEFAULT_USERNAME_IN_TEST = "user1"
    const val EXISTING_USERNAME = "~username~"

    private val userImageInTest = UserImage(byteArrayOf(1, 2, 3), MediaType.IMAGE_PNG_VALUE)

    val defaultUserProperties =
        DefaultUserProperties().apply {
            username = DEFAULT_USERNAME_IN_TEST
            password = "~default-user-password~"
            name = "~default-user-name~"
            email = "~default-user-email~"
        }

    fun customUserDetailsInTest(hasImage: Boolean = true) =
        CustomUserDetails(
            "~username~",
            "~password~",
            "~name~",
            "~email~",
            hasImage,
        )

    fun createUserInTest(userImage: UserImage? = userImageInTest) =
        VicxUser(
            EXISTING_USERNAME,
            VicxUser.VALID_BCRYPT_PASSWORD,
            "~name~",
            "~email~",
            userImage,
        )
}
